openedx/openedx-platform
# SE-1839 Allow overriding SEARCH_FILTER_GENERATOR

This allows setting a custom search filter generator used by edx-search. edx-search is already plugin based, but the search filter generator class is currently hardcoded for the lms. With this change, one can set a custom search filter class from a module/plugin.

**JIRA tickets**: [OSPR-3947](https://openedx.atlassian.net/browse/OSPR-3947)

**Dependencies**: None

**Sandbox URL**: 


-    LMS: https://pr22358.sandbox.opencraft.hosting/
-    Studio: https://studio.pr22358.sandbox.opencraft.hosting/



**Merge deadline**: None

**Testing instructions**:

1. set `ENABLE_COURSEWARE_SEARCH` and its friends.
1. reindex the courses to ensure results will actually be returned
1. Set a custom `SEARCH_FILTER_GENERATOR` (the sandbox is configured with the exclude_capa search filter from https://github.com/open-craft/opencraft-courseware-search-filters for testing)
1. run the lms
1. perform some  searches on the lms dashboard
1. verify that no CAPA items appear in search results (the configured search filter only does this on top of the default)
1. also should test that if `SEARCH_FILTER_GENERATOR` is *not* set, that the default fallback is correctly set. Reading the code should verify this.

**Author notes**:

**Reviewers**

**Settings**
```
EDXAPP_FEATURES:
  ENABLE_COURSEWARE_SEARCH: true
  ENABLE_DASHBOARD_SEARCH: true
  ENABLE_COURSEWARE_INDEX: true
  ENABLE_LIBRARY_INDEX: true
EDXAPP_LMS_ENV_EXTRA:
  SEARCH_FILTER_GENERATOR: 'opencraft_courseware_search_filters.ExcludeCapaSearchFilterGenerator'
EDXAPP_EXTRA_REQUIREMENTS:
  - name: 'git+https://github.com/open-craft/opencraft-courseware-search-filters.git@master#egg=opencraft_courseware_search_filters'
```
